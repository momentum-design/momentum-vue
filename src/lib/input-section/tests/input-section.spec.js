import { shallowMount } from '@vue/test-utils';
import InputSection from '../index.vue';

describe('InputSection', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(InputSection);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render input section with position class', () => {
    const wrapper = shallowMount(InputSection);

    expect(wrapper.find('.md-input__before').exists()).toEqual(true);
  });

  it('should render input section with position prop', () => {
    const wrapper = shallowMount(InputSection, {
      propsData: {
        position: 'after'
      }
    });

    expect(wrapper.find('.md-input__after').exists()).toEqual(true);
  });

  it('should render className if prop is passed', () => {
    const wrapper = shallowMount(InputSection, {
      attrs: {
        class: 'class-test'
      }
    });

    expect(wrapper.find('.class-test').exists()).toEqual(true);
  });

  it('should render children', () => {
    const wrapper = shallowMount(InputSection, {
      slots: {
        default: 'test'
      }
    });

    expect(wrapper.find('.md-input__before').text()).toEqual('test');
  });

  it('should pass otherProps to wrapper', () => {
    const wrapper = shallowMount(InputSection, {
      propsData: {
        id: 'testid'
      }
    });

    expect(wrapper.find('#testid').exists()).toEqual(true);
  });
});

