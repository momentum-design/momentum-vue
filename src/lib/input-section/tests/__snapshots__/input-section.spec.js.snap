// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSection should match snapshot 1`] = `<span class="md-input__before"></span>`;
